docs/docs/dev_docs/tutorials/token_portal/main.md

```diff
@@ -0,0 +1,62 @@
+
+A portal is the point of contact between L1 and a specific contract on Aztec. It allows for arbitrary message passing between L1 and Aztec, siloed just for the portal contract and its sister contract on Aztec. For applications such as token bridges, this is the point where the tokens are held on L1 while used in L2.
+
+### But why?
+
+Before portals, you had legos either just on L1 or just on L2. But for cross-chain, there was no arbitrary message passing bridge that didn’t introduce their own trust assumptions.
+
+Portals change this. With portals you can now have arbitrary execution across L1 and L2, paving the ground for seamless trustless composability with L1 and L2 legos, without introducing any trust assumptions across the L1 or Aztec network.
+
+That means your L1 application can have a sister application residing on L2 and both of them can work together across the two networks.
+
+<Image img={require("/img/tutorials/portals_shilling.png")} />
+
+### Cheap and private access to Ethereum
+
+Using portals, you could implement Aztec Connect-like functionality where you could deposit funds into a variety of DeFi protocols that reside on Ethereum even though your funds are on Aztec. This enables cheaper and private usage of existing dapps on Ethereum and prevents liquidity fragmentation.
+
+You could swap your L2 WETH into DAI on Uniswap on Ethereum and get the DAI on Aztec. Similarly, you could stake your L2 ETH into Lido on Ethereum and get stETH on Aztec!
+
+### L1\<\>L2 communication on Aztec
+
+Aztec has the following core smart contracts on L1 that we need to know about:
+
+- Rollup.sol - stores the current state of the rollup and includes logic to progress the rollup (i.e. the state transition function)
+- Inbox.sol - a mailbox to the rollup for L1 to L2 messages (e.g. depositing tokens). Portals put messages into the box, and the sequencers then decide which of these messages they want to include in their blocks, based on the inclusion fees they receive.
+- Outbox.sol - a mailbox to the rollup for L2 to L1 messages (e.g. withdrawing tokens). Aztec contracts emit these messages and the sequencer adds these to the outbox. Portals then consume these messages.
+- Registry.sol - just like L1, we assume there will be various versions of Aztec (due to upgrades, forks etc). In such a case messages must not be replayable in other Aztec “domains”. A portal must decide which version/ID of Aztec the message is for. The registry stores the rollup, inbox and outbox address for each version of Aztec deployments, so the portal can find out the address of the mailbox it wants to talk to
+
+For more information, read [cross-chain calls](https://docs.aztec.network/concepts/foundation/communication/cross_chain_calls).
+
+## Building a Token Bridge with Portals
+
+The goal for this tutorial is to create functionality such that a token can be bridged to and from Aztec. We’ll be using L1 to refer to Ethereum and L2 to refer to Aztec.
+
```

Comment:
Insert some generic legal disclaimer like "note this is just a reference implementation for demo/educational purposes only"